test/annotation_test.dart

```diff
@@ -221,12 +224,25 @@ void main() {
         expect(matched, isTrue);
       });
 
-      test('class annotated with type defined via package uri', () {
-        ClassElement annotatedClass =
-            _getAnnotationForClass(libElement, 'AnnotatedWithJson');
-        var annotation = annotatedClass.metadata.single;
-        var matched = matchAnnotation(JsonSerializable, annotation);
-        expect(matched, isTrue);
+      group('class annotated with type defined via package uri', () {
+        test('When running on a normal file system', () {
+          ClassElement annotatedClass =
+              _getAnnotationForClass(libElement, 'AnnotatedWithJson');
+          var annotation = annotatedClass.metadata.single;
+          var matched = matchAnnotation(JsonSerializable, annotation);
+          expect(matched, isTrue);
+        });
+
+        test('When running in barback', () {
+          var annotation = _mockElementAnnotation(
```

Comment:
Ya, its definitely brittle. All the other ways I can think of to test this are a lot worse though, and would require a barback dependency and mocking out a bunch of barback stuff :/